internal/agent/service/report.go

```diff
@@ -64,6 +72,66 @@ func (r *report) Send() {
 
 func (r *report) MassSend() error {
 	const retryCount = 3
+	metrics := r.getMetrics()
+
+	r.Storage.ClearCounter()
+
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	action := func() error {
+		return r.request(metrics, r.massUpdateURL)
+	}
+
+	if err := retry.New(action, r.isConnRefused, retryCount).Run(); err != nil {
+		return fmt.Errorf("massSend(): failed to send metrics, %w", err)
+	}
+
+	return nil
+}
+
+func (r *report) PoolSend() {
+	metrics := r.getMetrics()
+	numJobs := len(metrics)
+	jobs := make(chan model.Metric, numJobs)
+	results := make(chan int, numJobs)
+
+	r.Storage.ClearCounter()
+
+	for w := 1; w <= r.rateLimit; w++ {
```

Comment:
Так более идиоматично:

```suggestion
	for w := 0; w < r.rateLimit; w++ {
```